code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1750,4 +1750,21 @@
 
 	hud_set_squad()
 
-	return TRUE
\ No newline at end of file
+	return TRUE
+
+/mob/living/carbon/human/screech_act(mob/living/carbon/Xenomorph/Queen/Q)
+	shake_camera(src, 3 SECONDS, 1) //50 deciseconds, SORRY 5 seconds was way too long. 3 seconds now
+	var/dist = get_dist(src, Q)
+	var/reduction = max(1 - 0.1 * protection_aura, 0) //Hold orders will reduce the Halloss; 10% per rank.
+	var/halloss_damage = (max(0,140 - dist * 10)) * reduction //Max 130 beside Queen, 70 at the edge
+	var/stun_duration = max(0,1.1 - dist * 0.1) * reduction //Max 1 beside Queen, 0.4 at the edge.
+
+	if(dist < 8)
```

Comment:
Do an early return here.
Basically if the contrary, then return.
Such as
```
if(dist > 7)
	return
```
Then you'll have to adjust the indentation of the rest.

Reply:
Can this just be removed as we rely on `range(world.view)` instead?

Reply:
It's more of a sanity check. Ideally, yes, there's no need for this.
Maybe do a
```
if(dist > 7)
	stack_trace("[src] called `screech_act()` with [dist] dist to the source of the screech, [Q]")
	return
```
That will inform us if it actually happens due to lag or something.
The problem with sanity checks is when they are silent, for the most part.